SandWorm/SandWormSetupComponent.cs

```diff
@@ -0,0 +1,234 @@
+﻿using System;
+using System.Collections.Generic;
+using Grasshopper.Kernel;
+using System.Linq;
+
+namespace SandWorm
+
+{
+    public class SandWormSetupComponent : GH_Component
+    {
+        public bool calibrateSandworm;
+        public double sensorElevation = 1000; // Arbitrary default value (must be >0)
+        public int leftColumns = 0;
+        public int rightColumns = 0;
+        public int topRows = 0;
+        public int bottomRows = 0;
+        public int tickRate = 33; // In ms
+        public int keepFrames = 1; // In ms
+
+        public int frameCount; // Number of frames to average the calibration across
+        public double averagedSensorElevation;
+        public double[] elevationArray = Enumerable.Range(1, 217088).Select(i => new double()).ToArray();
+        public int[] runningSum = Enumerable.Range(1, 217088).Select(i => new int()).ToArray();
+        private LinkedList<int[]> renderBuffer = new LinkedList<int[]>();
+
+        /// <summary>
+        /// Initializes a new instance of the MyComponent1 class.
+        /// </summary>
+        public SandWormSetupComponent()
+          : base("SandWormSetup", "SWSetup",
+              "This component takes care of all the setup & calibration of your sandbox.",
+              "Sandworm", "Sandbox")
+        {
+        }
+
+        /// <summary>
+        /// Registers all the input parameters for this component.
+        /// </summary>
+        protected override void RegisterInputParams(GH_Component.GH_InputParamManager pManager)
+        {
+            pManager.AddBooleanParameter("CalibrateSandworm", "CS", "Set to true to initiate the calibration process.", GH_ParamAccess.item, calibrateSandworm);
```

Comment:
Could we add a note here about running calibrating on a flat table? I imagine that any official documentation would run through this process explicitly, but I think its a good idea to try and provide 'fallback' help within Grasshopper itself.

Reply:
Good idea.